compiler/AST/type.cpp

```diff
@@ -1166,6 +1172,99 @@ Symbol* AggregateType::getField(int i) {
   return toDefExpr(fields.get(i))->sym;
 }
 
+
+void AggregateType::printDocs(std::ostream *file, unsigned int tabs) {
+  if (this->symbol->hasFlag(FLAG_NO_DOC) || this->isUnion()) {
+    return;
+  }
+
+  this->printTabs(file, tabs);
+  *file << this->docsDirective();
+  *file << this->symbol->name;
+  *file << this->docsSuperClass();
+  *file << std::endl;
+
+  // In rst mode, ensure there is an empty line between the class/record
+  // signature and its description or the next directive.
+  if (!fDocsTextOnly) {
+    *file << std::endl;
+  }
+
+  if (this->doc != NULL) {
+    this->printDocsDescription(this->doc, file, tabs + 1);
+    *file << std::endl;
+
+    // In rst mode, ensure there is an empty line between the class/record
+    // description and the next directive.
+    if (!fDocsTextOnly) {
+      *file << std::endl;
+    }
+  }
+}
+
+
+/*
+ * Returns super class string for documentation. If super class exists, returns
+ * ": <super class name>".
+ */
+std::string AggregateType::docsSuperClass() {
+  if (this->inherits.length > 0) {
+    std::vector<std::string> superClassNames;
+
+    for_alist(expr, this->inherits) {
+      if (UnresolvedSymExpr* use = toUnresolvedSymExpr(expr)) {
+        superClassNames.push_back(use->unresolved);
+      }
+    }
+
+    if (superClassNames.empty()) {
```

Comment:
I think there should be an internal error/assert if the inheritance list contains something that we don't expect, which would cause this check to result in true, since it would mean something went wrong during parsing on our end.

Additionally, by not looking up the parent, we're losing some checks which would otherwise occur (records inheriting from classes and vice versa, trying to inherit from a parent that doesn't exist, etc.).  I don't think these checks are the responsibility of the docs pass, but at least the case of inheriting from a non-existent would cause problems for us in documentation generation should it occur.  Should we have a way of detecting that it happened, either here or in the Sphinx domain?